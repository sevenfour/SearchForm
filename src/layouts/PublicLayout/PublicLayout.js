import React, { Suspense } from 'react';
import PropTypes from 'prop-types';

import { renderRoutes } from 'react-router-config';

import { useProgressProviderContext } from 'context/ProgressContextProvider';

import Loading from 'components/Loading';

import styles from './PublicLayout.module.css';

const loader = (
  <div className={styles.loadingContainer}>
    <Loading overlay />
  </div>
);

const PublicLayout = props => {

  const { route } = props;

  const { loading } = useProgressProviderContext();

  return (
    <>
      <header />
      <main
        aria-busy={loading ? true : false}
        aria-live="polite"
        className={styles.container}
      >
        <Suspense fallback={loader}>
          {renderRoutes(route.routes)}
        </Suspense>
      </main>
      <footer />
    </>
  );
};

PublicLayout.propTypes = {
  route: PropTypes.object
};

export default PublicLayout;
