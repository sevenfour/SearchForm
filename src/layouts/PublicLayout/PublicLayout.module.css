
.container {
  max-width: 1296px;
  height: 100vh;
}

.loadingContainer {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}
