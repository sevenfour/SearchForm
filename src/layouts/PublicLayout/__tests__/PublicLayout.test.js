import React from 'react';
import { shallow } from 'enzyme';

import PublicLayout from '..';

describe('<PublicLayout /> component tests', () => {

  const props = {
    route: {
      routes: [
        {
          path: '/',
          exact: true,
          component: () => {}
        }
      ]
    }
  };

  it('renders', () => {
    const publicLayout = shallow(
      <PublicLayout
        {...props}
      />
    );
    expect(publicLayout).toMatchSnapshot();
  });
});
