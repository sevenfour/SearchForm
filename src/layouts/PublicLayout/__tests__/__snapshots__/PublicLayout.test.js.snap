// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PublicLayout /> component tests renders 1`] = `
<Fragment>
  <header />
  <main>
    <Suspense
      fallback={
        <div
          className="loadingContainer"
        >
          <Loading />
        </div>
      }
    >
      <Switch>
        <Route
          exact={true}
          key="0"
          path="/"
          render={[Function]}
        />
      </Switch>
    </Suspense>
  </main>
  <footer />
</Fragment>
`;
